### ISSUE TITLE
[pilx] Exception when running the diffmap

### ISSUE THREAD

**Issue by kif** [silx-kit/pyFAI#2142]
```
ERROR:silx.gui.plot.tools.PositionInfo:Error while converting coordinates (81.000000, 57.000000)with converter 'Data'
ERROR:silx.gui.plot.tools.PositionInfo:Traceback (most recent call last):
  File "/users/kieffer/.venv/py312/lib/python3.12/site-packages/silx/gui/plot/tools/PositionInfo.py", line 299, in _updateStatusBar
    value = func(xData, yData)
            ^^^^^^^^^^^^^^^^^^
  File "/users/kieffer/.venv/py312/lib/python3.12/site-packages/pyFAI/gui/pilx/widgets/MapPlotWidget.py", line 94, in _dataConverter
    index = self.getScatterIndex(x, y)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/users/kieffer/.venv/py312/lib/python3.12/site-packages/pyFAI/gui/pilx/widgets/MapPlotWidget.py", line 168, in getScatterIndex
    pixel_x, pixel_y = self.dataToPixel(x_data, y_data)
    ^^^^^^^^^^^^^^^^
TypeError: cannot unpack non-iterable NoneType object
```

Should at least be silenced out.

**Comment by loichuder** [comment]
How to reproduce this ?

**Comment by EdgarGF93** [comment]
It happens sometimes when the mouse is right at the edge of the map widget. Due to numeric precision, sometimes the position of the cursor goes out of the edges.

I put these debugging lines at `datatoPixel` method
```

        if check:
            isOutside = numpy.logical_or(
                numpy.logical_or(x > xmax, x < xmin),
                numpy.logical_or(y > ymax, y < ymin),
            )
            print(f"{isOutside=}")
            print(f"{xmin=}, {xmax=}")
            print(f"{ymin=}, {ymax=}")


```
```
xmin=5.2634029056681, xmax=5.406640065479845
ymin=1.3077485380116958, ymax=2.279970760233918
x is 5.388456244332473, y is 2.273148148148148
isOutside=False
xmin=5.2634029056681, xmax=5.406640065479845
ymin=1.3077485380116958, ymax=2.279970760233918
x is 5.385642500247774, y is 2.2816764132553606
isOutside=True
xmin=5.2634029056681, xmax=5.406640065479845
ymin=1.3077485380116958, ymax=2.279970760233918
ERROR:silx.gui.plot.tools.PositionInfo:Error while converting coordinates (429.000000, 30.000000)with converter 'Data'
ERROR:silx.gui.plot.tools.PositionInfo:Traceback (most recent call last):
  File "/users/edgar1993a/anaconda3/envs/ewoks/lib/python3.9/site-packages/silx/gui/plot/tools/PositionInfo.py", line 299, in _updateStatusBar
    value = func(xData, yData)
  File "/home/edgar1993a/work/pyFAI/build/lib/python3.9/site-packages/pyFAI/gui/pilx/widgets/MapPlotWidget.py", line 104, in _dataConverter
    index = self.getScatterIndex(x, y)
  File "/home/edgar1993a/work/pyFAI/build/lib/python3.9/site-packages/pyFAI/gui/pilx/widgets/MapPlotWidget.py", line 182, in getScatterIndex
    pixel_x, pixel_y = self.dataToPixel(x_data, y_data)
TypeError: cannot unpack non-iterable NoneType object

```

**Comment by EdgarGF93** [comment]
fixed in #2235

**Comment by EdgarGF93** [comment]
It happens only when the cursor is moving around the left and top axis (x-minimum and y-maximum). In this patch, I catch the error and continue.
